fairing/preprocessors/converted_notebook.py

```diff
@@ -65,16 +65,20 @@ def __init__(self,
 
         self.notebook_file = notebook_file
         self.notebook_preprocessor = notebook_preprocessor
+        self._not_overwrite = False
 
     def preprocess(self):
+        if self._not_overwrite:
+            return [self.executable]
+        self._not_overwrite = True
         exporter = nbconvert.PythonExporter()
         exporter.register_preprocessor(self.notebook_preprocessor, enabled=True)
         contents, _ = exporter.from_filename(self.notebook_file)
         converted_notebook = Path(self.notebook_file).with_suffix('.py')
         if converted_notebook.exists():
-            dir_name = Path(self.notebook_file).parent
+            dir_name = Path(self.notebook_file).absolute().parent
```

Comment:
We still have two different code paths  i.e. the if block will not be executed if the file doesn't already exist.

I think the desired behavior is

1. Check if file exists
1. If file exists and overwrite=False; raise exception
1. Otherwise overwrite the file

In either case we should always use the same path for the file we write to.